aries/aries_vcx/src/common/signing.rs

```diff
@@ -146,3 +152,22 @@ pub async fn decode_signed_connection_response(
 //         .await;
 //     }
 // }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use base64::{self, engine::general_purpose, Engine};
+
+    #[tokio::test]
+    async fn test_decode_padded_and_unpadded_sig_data() {
+        let data = b"example data for testing";
+        let encoded_padded = general_purpose::URL_SAFE.encode(data);
+        let encoded_unpadded = general_purpose::URL_SAFE_NO_PAD.encode(data);
+
+        let decoded_padded = base64url_decode(&encoded_padded).unwrap();
+        let decoded_unpadded = base64url_decode(&encoded_unpadded).unwrap();
+
+        assert_eq!(decoded_padded, data.to_vec());
+        assert_eq!(decoded_unpadded, data.to_vec());
+    }
+}
```

Comment:
nit; for completeness... could you pls test the error case too? (checking that `InvalidJson` is thrown)